kiva/agg/agg-24/font_freetype/agg_font_freetype.cpp

```diff
@@ -567,12 +569,24 @@ namespace agg24
 
 
     //------------------------------------------------------------------------
-    int font_engine_freetype_base::find_face(const char* face_name) const
+    int font_engine_freetype_base::find_face(const char* face_name, unsigned face_name_len, unsigned face_index)
     {
         unsigned i;
+
+        if ((face_name_len + 4 + 1) > m_face_lookup_scratch_len)

```

Comment:
i'm not entirely sure what the magical 4 and 1 here are.

Reply:
4 for the face index (this is the `%04u` in the format string) and 1 for the NULL.